Update proof of stake section (wiki migration) (#6157)
ethereum/ethereum-org-website: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md

---
title: Gasper

> description: An explanation of the Gasper PoS mechanism.

Reviewer: ```suggestion
description: An explanation of the Gasper proof-of-stake mechanism.
```

lang: en
sidebar: true
incomplete: false
---

Gasper is a combination of Casper the Friendly Finality Gadget and the LMD-GHOST fork choice algorithm. Together these components form the consensus mechanism securing proof-of-stake Ethereum. Casper is the mechanism that uprgades certain blocks to "finalized" so that new entrants into the network can be confident that they are syncing the canonical chain. The fork choice algorithm uses accumulated votes to ensure that when forks arise in the blockchain nodes can easily select the correct one.

> 

Author: done

**note** that the original definition of Casper-FFG was updated slightly for inclusion in Gasper. On this page we consider the updated version.

> ## The role of Gasper

Reviewer: Could you please add header ids to any new headings 🙏
```suggestion
## The role of Gasper {#gasper}
```

Author: done

---
title: Gasper
description: An explanation of the Gasper proof-of-stake mechanism.
lang: en
sidebar: true
incomplete: false
---

Gasper is a combination of Casper the Friendly Finality Gadget (Casper-FGG) and the LMD-GHOST fork choice algorithm. Together these components form the consensus mechanism securing proof-of-stake Ethereum. Casper is the mechanism that upgrades certain blocks to "finalized" so that new entrants into the network can be confident that they are syncing the canonical chain. The fork choice algorithm uses accumulated votes to ensure that nodes can easily select the correct one when forks arise in the blockchain.

> **note** that the original definition of Casper-FFG was updated slightly for inclusion in Gasper. On this page we consider the updated version.

Reviewer 2: ```suggestion
**Note** that the original definition of Casper-FFG was updated slightly for inclusion in Gasper. On this page we consider the updated version.
```

## The role of Gasper {#role-of-gasper}

Gasper sits on top of a proof-of-stake blockchain where nodes provide ether as a security deposit that can be destroyed if they are lazy or dishonest in proposing or validating blocks. Gasper is the mechanism defining how validators get rewarded and punished, decide which blocks to accept and reject, and which fork of the blockchain to build on.

## What is finality? {#what-is-finality}

> [Casper the Friendly Finality Gadget (Casper-FFG)](https://arxiv.org/pdf/1710.09437.pdf) is an algorithm that finalizes blocks. Finalizing a block means upgrading a block so that it cannot be reverted (unless there has been a critical consensus failure). Finalized blocks can be thought of as information the blockchain is certain about. A block must pass through a two-step upgrade procedure for a block to be finalized:

Reviewer 2: Small note, but feel like we should open this section with a sentence directly answering "What is finality?" before explaining more about Casper, etc.

Author: yup - done

Because finality requires a two-thirds agreement that a block is canonical, an attacker cannot possibly create an alternative finalized chain without: 

1. Owning or manipulating two-thirds of the total staked ether.
2. Destroying at least one-third of the total staked ether. 

The first condition arises because two-thirds of the staked ether is required to finalize a chain. The second condition arises because if two-thirds of the total stake has voted in favor of both forks, then one-third must have voted on both. Double-voting is a slashing condition that would be maximally punished, and one-third of the total stake would be destroyed. As of May 2022, this requires an attacker to burn around $10 billion worth of ether.

### Incentives and Slashing {#incentives-and-slashing}

> Validators get rewarded for honestly proposing and validating blocks. The rewards ether added to their stake. On the other hand, validators that are absent and fail to act when called upon miss out on these rewards and sometimes lose a small portion of their existing stake. However, the penalties for being offline are small and, in most cases, amount to opportunity costs of missing rewards. However, some validator actions are very difficult to do accidentally and signify some malicious intent, such as proposing multiple blocks for the same slot, attesting to multiple blocks for the same slot, or contradicting previous checkpoint votes. These are "slashable" behaviors that are penalized more harshly—slashing results in some portion of the validator's stake being destroyed and the validator being removed from the network. This process takes 36 days. On Day 1, there is an initial penalty of up to 0.5 ETH. Then the slashed validator's ether slowly drains away across the exit period, but on Day 18, they receive a "correlation penalty", which is larger when more validators are slashed around the same time. The maximum penalty is the entire stake. These rewards and penalties are designed to incentivize honest validators and disincentivize attacks on the network.

Reviewer 2: >  The rewards ether added to their stake. 

Attempt at correcting:
```suggestion
Validators get rewarded for honestly proposing and validating blocks. Ether is rewarded and added to their stake. On the other hand, validators that are absent and fail to act when called upon miss out on these rewards and sometimes lose a small portion of their existing stake. However, the penalties for being offline are small and, in most cases, amount to opportunity costs of missing rewards. However, some validator actions are very difficult to do accidentally and signify some malicious intent, such as proposing multiple blocks for the same slot, attesting to multiple blocks for the same slot, or contradicting previous checkpoint votes. These are "slashable" behaviors that are penalized more harshly—slashing results in some portion of the validator's stake being destroyed and the validator being removed from the network of validators. This process takes 36 days. On Day 1, there is an initial penalty of up to 0.5 ETH. Then the slashed validator's ether slowly drains away across the exit period, but on Day 18, they receive a "correlation penalty", which is larger when more validators are slashed around the same time. The maximum penalty is the entire stake. These rewards and penalties are designed to incentivize honest validators and disincentivize attacks on the network.
```

---
title: Gasper
description: An explanation of the Gasper proof-of-stake mechanism.
lang: en
sidebar: true

> incomplete: false

Reviewer 3: ✂️ - see comment below.
